File: Learning excercises/CSharpSnapIn/Class1.cs
using CommonSnappableTypes;
using System.Windows.Forms;

namespace CSharpSnapIn
{
    [CompanyInfo(CompanyName ="CSharp company",CompanyUrl ="www.yandex.ru")]
    public class CSharpModule:IPublicFuncionality
    {
        void IPublicFuncionality.DoIt() => MessageBox.Show("C Sharp module");
    }
}

File: Learning excercises/VBSnapIn/Class1.vb
Imports System.Windows.Forms
Imports CommonSnappableTypes

<CompanyInfo(CompanyName:="Chucky bear", CompanyUrl:="google.com")>
Public Class VBSnapIn
    Implements IPublicFuncionality

    Public Sub DoIt() Implements IPublicFuncionality.DoIt
        MessageBox.Show("Snap from Visual basic")
    End Sub
End Class
